### PR TITLE
feat: make event communication more reliable 

### PR DESCRIPTION
**Pull request by kamilbaczek**
PR includes:
- Eliminating the event bus abstraction, as MassTransit already serves as a comprehensive abstraction. The current IEventBus only hinders the utilization of MassTransit's full capabilities.
- Implementation of the Inbox Pattern.
- Implementation of the Outbox Pattern.
- Introduction of Retries and Redelivery mechanisms.
- Adoption of ADRs (Architectural Decision Records).

Not includes:
- bump common infrastructure versions
- retries and redelivery values from configuration instead static

### CODE REVIEW

File: Chapter-3-microservice-extraction/Fitnet/Src/Fitnet/appsettings.Development.json [hunk 1]
```diff
@@ -13,7 +13,6 @@
   },
   "ConnectionStrings": {
     "Passes": "Host=postgres:5432;Database=fitnet;Username=postgres;Password=mysecretpassword",
-    "Contracts": "Host=postgres:5432;Database=fitnet;Username=postgres;Password=mysecretpassword",
```
Comment (kamilbaczek) [inline Chapter-3-microservice-extraction/Fitnet/Src/Fitnet/appsettings.Development.json:16]: In modular monolith app this config is not relevent.

File: Chapter-3-microservice-extraction/Docs/ArchitectureDecisionLog/0023-implementation-of-Inbox-Outbox-Pattern-with-MassTransit.adoc [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+- *Scalability:* MassTransit's inbox-outbox pattern is designed to scale efficiently. This scalability is crucial for our system as it grows, ensuring that our messaging architecture can handle increasing loads without compromising performance.
+
+- *Learning Curve:*
+Teams unfamiliar with MassTransit may face a learning curve in understanding the intricacies of the framework. Training and onboarding efforts might be required to ensure effective utilization.
+
+- *Dependency on External Framework:*
+Adopting MassTransit means introducing a dependency on an external framework. This may pose challenges if the framework evolves or undergoes significant changes that require adjustments in our system.
+
+- *Limited Customization:*
```
Comment (kamilbaczek) [inline Chapter-3-microservice-extraction/Docs/ArchitectureDecisionLog/0023-implementation-of-Inbox-Outbox-Pattern-with-MassTransit.adoc:39]: add reference to MassTransit doc

File: Chapter-3-microservice-extraction/Fitnet.Contracts/Src/Fitnet.Contracts/Properties/launchSettings.json [hunk 3]
```diff
@@ -14,7 +14,7 @@
       "dotnetRunMessages": true,
       "launchBrowser": true,
       "launchUrl": "swagger",
-      "applicationUrl": "http://localhost:5013",
+      "applicationUrl": "http://localhost:5010",
```
Comment (kamilbaczek) [inline Chapter-3-microservice-extraction/Fitnet.Contracts/Src/Fitnet.Contracts/Properties/launchSettings.json:17]: url has changed to easly run multiple services

File: Chapter-3-microservice-extraction/Fitnet/Src/Passes/Fitnet.Passes.Api/Common/EventBus/EventBusModule.cs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    internal static IServiceCollection AddEventBus(this IServiceCollection services, IConfiguration configuration)
+    {
+        services.Configure<EventBusOptions>(options => configuration.GetSection(EventBusConfiguration).Bind(options));
+        services.AddMassTransit(configurator =>
+        {
+            configurator.AddEntityFrameworkOutbox<PassesPersistence>(o =>
+            {
+                o.UsePostgres();
+                o.UseBusOutbox();
+                o.DuplicateDetectionWindow = TimeSpan.FromSeconds(30);
```
Comment (kamilbaczek) [inline Chapter-3-microservice-extraction/Fitnet/Src/Passes/Fitnet.Passes.Api/Common/EventBus/EventBusModule.cs:23]: should be in json configuration as well

File: Chapter-3-microservice-extraction/Fitnet/Src/Passes/Fitnet.Passes.Api/Common/EventBus/EventBusModule.cs [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+internal static class EventBusModule
+{
+    private const string EventBusConfiguration = "EventBus";
+
+    internal static IServiceCollection AddEventBus(this IServiceCollection services, IConfiguration configuration)
+    {
+        services.Configure<EventBusOptions>(options => configuration.GetSection(EventBusConfiguration).Bind(options));
+        services.AddMassTransit(configurator =>
+        {
+            configurator.AddEntityFrameworkOutbox<PassesPersistence>(o =>
```
Comment (kamilbaczek) [inline Chapter-3-microservice-extraction/Fitnet/Src/Passes/Fitnet.Passes.Api/Common/EventBus/EventBusModule.cs:19]: Can be extracted to private method

File: Chapter-3-microservice-extraction/Fitnet.Contracts/Src/Fitnet.Contracts.IntegrationEvents/ContractSignedEvent.cs [hunk 6]
```diff
@@ -1,9 +1,7 @@
 namespace EvolutionaryArchitecture.Fitnet.Contracts.IntegrationEvents;
 
-using Common.Infrastructure.Events;
-
-public sealed record ContractSignedEvent(Guid Id, Guid ContractId, Guid ContractCustomerId, DateTimeOffset SignedAt, DateTimeOffset ExpireAt, DateTimeOffset OccurredDateTime) : IIntegrationEvent
+public sealed record ContractSignedEvent(Guid Id, Guid ContractId, Guid ContractCustomerId, DateTimeOffset SignedAt, DateTimeOffset ExpireAt, DateTimeOffset OccurredDateTime)
 {
-    public static ContractSignedEvent Create(Guid contractId, Guid contractCustomerId, DateTimeOffset signedAt, DateTimeOffset expireAt) =>
+    public static ContractSignedEvent ````Create(Guid contractId, Guid contractCustomerId, DateTimeOffset signedAt, DateTimeOffset expireAt) =>
```
Comment (meaboutsoftware) [inline Chapter-3-microservice-extraction/Fitnet.Contracts/Src/Fitnet.Contracts.IntegrationEvents/ContractSignedEvent.cs:5]: Remove ````

File: Chapter-3-microservice-extraction/Fitnet/Src/Fitnet/appsettings.json [hunk 7]
```diff
@@ -1,7 +1,7 @@
 {
   "Logging": {
     "LogLevel": {
-      "Default": "Information",
+      "Default": "Debug",
```
Comment (meaboutsoftware) [inline Chapter-3-microservice-extraction/Fitnet/Src/Fitnet/appsettings.json:4]: Why debug in appsettings? Appsettings.json is considered as production ready configuration (even if not used) and thats why Logging level is set to information

File: Chapter-3-microservice-extraction/Fitnet/Src/Passes/Fitnet.Passes.IntegrationEvents/Fitnet.Passes.IntegrationEvents.csproj [hunk 8]
```diff
@@ -1,5 +1,8 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <ItemGroup>
-    <PackageReference Include="evolutionaryarchitecture.fitnet.common.infrastructure" Version="1.1.5" />
+    <PackageReference Include="MassTransit.RabbitMQ" Version="8.1.1" />
+  </ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include="..\..\..\..\Fitnet.Common\Fitnet.Common.Infrastructure\Fitnet.Common.Infrastructure.csproj" />
```
Comment (meaboutsoftware) [inline Chapter-3-microservice-extraction/Fitnet/Src/Passes/Fitnet.Passes.IntegrationEvents/Fitnet.Passes.IntegrationEvents.csproj:6]: Why did you change it to project reference of a relative path instead of 1.1.7 package?

File: Chapter-3-microservice-extraction/Fitnet/Src/Reports/Fitnet.Reports/Fitnet.Reports.csproj [hunk 9]
```diff
@@ -3,12 +3,13 @@
         <FrameworkReference Include="Microsoft.AspNetCore.App" />
     </ItemGroup>
     <ItemGroup>
-      <PackageReference Include="Dapper" Version="2.0.123" />
+      <PackageReference Include="Dapper" Version="2.1.15" />
       <PackageReference Include="evolutionaryarchitecture.fitnet.common.api" Version="1.1.5" />
```
Comment (meaboutsoftware) [inline Chapter-3-microservice-extraction/Fitnet/Src/Reports/Fitnet.Reports/Fitnet.Reports.csproj:7]: Would be great to update all packages to the newest version (can be done in the other PR)

**Review by meaboutsoftware — CHANGES_REQUESTED** [review]
We can think bout adding GlobalUsings.cs to each project - this way it will be more readable :)

File: Chapter-3-microservice-extraction/Fitnet/Src/Fitnet/appsettings.json (hunk 7 above)
Comment (kamilbaczek) [reply Chapter-3-microservice-extraction/Fitnet/Src/Fitnet/appsettings.json]: Yeah, Fixed

File: Chapter-3-microservice-extraction/Fitnet/Src/Reports/Fitnet.Reports/Fitnet.Reports.csproj (hunk 9 above)
Comment (kamilbaczek) [reply Chapter-3-microservice-extraction/Fitnet/Src/Reports/Fitnet.Reports/Fitnet.Reports.csproj:7]: I'm going to do it in dotnet upgrade

File: Chapter-3-microservice-extraction/Fitnet/Src/Passes/Fitnet.Passes.IntegrationEvents/Fitnet.Passes.IntegrationEvents.csproj (hunk 8 above)
Comment (kamilbaczek) [reply Chapter-3-microservice-extraction/Fitnet/Src/Passes/Fitnet.Passes.IntegrationEvents/Fitnet.Passes.IntegrationEvents.csproj]: That was for local testing. I've reverted.

File: Chapter-3-microservice-extraction/Fitnet/Src/Passes/Tests/Fitnet.Passes.IntegrationTests/RegisterPass/RegisterPassTests.cs [hunk 10]
```diff
@@ -33,12 +33,5 @@ internal async Task Given_contract_signed_event_Then_should_register_pass()
 
         // Assert
         _testEventBus.EnsureConsumed<ContractSignedEvent>();
-        await EnsureThatPassRegistered();
-    }
-
-    private async Task EnsureThatPassRegistered()
```
Comment (kamilbaczek) [inline Chapter-3-microservice-extraction/Fitnet/Src/Passes/Tests/Fitnet.Passes.IntegrationTests/RegisterPass/RegisterPassTests.cs:39]: https://github.com/evolutionary-architecture/evolutionary-architecture-by-example/issues/45
Issue Created to restore this change
